taker-frontend/src/components/Trade.tsx

```diff
@@ -101,6 +105,47 @@ const Trade = (
         },
     });
 
+    const parse = (val: any) => Number.parseInt(val.replace(/^\$/, ""));
+
+    const balanceTooLow = walletBalance && walletBalance < parse(margin);
+    const quantityTooHigh = maxQuantity < parse(quantity);
+    const quantityTooLow = minQuantity > parse(quantity);
+    const quantityGreaterZero = parse(quantity) > 0;
+
+    const canSubmit = orderId && !isSubmitting && !balanceTooLow
+        && !quantityTooHigh && !quantityTooLow && quantityGreaterZero;
+
+    let alertBox;
+
+    if (balanceTooLow) {
+        alertBox = <Alert status="error">
+            <AlertIcon />
+            <AlertTitle mr={2}>Your balance is too low!</AlertTitle>
+            <AlertDescription>Pleas deposit more into you wallet.</AlertDescription>
+        </Alert>;
+    }
+    if (quantityTooHigh) {
+        alertBox = <Alert status="error">
+            <AlertIcon />
+            <AlertTitle mr={2}>Quantity too high!</AlertTitle>
+            <AlertDescription>Max available liquidity is ${maxQuantity}.</AlertDescription>
+        </Alert>;
+    }
+    if (quantityTooLow || !quantityGreaterZero) {
+        alertBox = <Alert status="error">
+            <AlertIcon />
+            <AlertTitle mr={2}>Quantity too low!</AlertTitle>
+            <AlertDescription>Min quantity is ${minQuantity}.</AlertDescription>
+        </Alert>;
+    }
+    if (!orderId) {
+        alertBox = <Alert status="error">
+            <AlertIcon />
+            <AlertTitle mr={2}>No liquidity!</AlertTitle>
+            <AlertDescription>The maker you are connected to did not create any offers.</AlertDescription>
```

Comment:
nit: "has not created"? Present perfect tense might be slightly better

Reply:
done :)